### DISCUSSION TITLE
Connect Application to DB from another ServiceStack

### DISCUSSION THREAD

**Discussion by m-stokowski** [coollabsio/coolify#1803] [Q&A]
Hi,
I'm trying to connect an App to DB from another service stack on the same server without making DB accessible over the internet.

My config looks like this:
- service stack "A" based on docker-composefile with 2 containers: "app1" and "mysql" in it. DB container exposes port 3306.
- network "netA" that was created automatically by Coolify with service stack "A" - I've added "netA" to Coolify server destinations.
- "app2" built with Nixpack that is set-up to connect to network/destination "netA" created by previous stack.

I would like to connect "app2" to DB from service stack "A" ("mysql" container) but every time i try to deploy "app2" build process fails when it tries to pull data from DB.

In environment variables of "app2" i tried setting DB_HOST to:
- mysql
- mysql-netA (container name with uuid set by default by coolify. Uuid is the same as network name)
- host.docker.internal

DB_PORT is set to 3306.

What am I doing wrong and what is the proper way to connect both containers?

**Comment by softwaretirol** [comment]
Same question here...

**Comment by andrasbacsai** [comment]
If you are using `Nixpacks` build pack, the db needs to be publicly available unfortunately. There is an [open issue](https://github.com/railwayapp/nixpacks/issues/1057) at Nixpacks, that will probably solve the issue.

Another workaround is to use Dockerfile based deployment.

**Reply by ApoloApps** [reply]
so this means i cant use postgres internal url with uptime kuma? im having problems

**Comment by mutonby** [comment]
and with a docker compose? what internal dns can I use? localhost not working mapping the ports

**Reply by mk3-14159** [reply]
Any news on this?

**Comment by alocin98** [comment]
I am using 172.17.0.1 since this seems to always be the docker host's IP Address. Just spent hours finding this out.

**Comment by Cristy94** [comment]
So, there is no way to connect to a DB in another stack, without making the DB publicly available?

**Reply by djsisson** [reply]
there is, you just need to make sure both stacks are in the same docker network as the db, usually you would place your db in its own network, and then for each service that needs to access it, you would add that one service to the same network as your db

please note coolify does have issues with containers in multiple networks when using traefik, but it can be resolved manually

or you make the db 'privately available' i.e only internally to your host, not 5432:5432 but 172,17,0,1:5432:5432 then use host.docker.internal:5432 (using your docker bridge ip) you can do the same if you have a lan ip
